lnwallet/channel.go

```diff
@@ -2527,9 +2558,10 @@ func (lc *LightningChannel) evaluateHTLCView(view *htlcView, ourBalance,
 		// and it hasn't been processed, yet, the increment our state
 		// tracking the total number of satoshis we've sent within the
 		// channel.
-		if mutateState && entry.EntryType == Settle && !remoteChain &&
+		if entry.EntryType == Settle && !remoteChain &&
 			entry.removeCommitHeightLocal == 0 {
-			lc.channelState.TotalMSatSent += entry.Amount
+
+			evaluation.sentTotal += entry.Amount
```

Comment:
I like where this is heading! Could the same be done for the HTLC mutation? That being everywhere we now check `mutateState`, add the mutation to the `viewEvaluation` (could even be a list of `func`'s to run when we mutate), then evaluate this outside `evaluateView`. Feels like that would significantly reduce the diff ...